src/System.Text.Primitives/System/Text/Parsing/InvariantSigned.cs

```diff
@@ -899,176 +899,89 @@ public unsafe static bool TryParseInt32(byte* text, int length, out int value, o
-                            value = parsedValue * sign;
-                            return true;
-                        }
-                        parsedValue = parsedValue * 10 + nextDigit;
-                    }
-                }
-                else
-                {
-                    // Length is greater than Int32OverflowLength; overflow is only possible after Int32OverflowLength
-                    // digits. There may be no overflow after Int32OverflowLength if there are leading zeroes.
-                    for (int index = indexOfFirstDigit + 1; index < overflowLength - 1; index++)
-                    {
-                        int nextDigit = text[index] - 48; // '0'
-                        if (nextDigit < 0 || nextDigit > 9)
-                        {
-                            value = parsedValue * sign;
-                            return true;
-                        }
-                        parsedValue = parsedValue * 10 + nextDigit;
-                    }
-                    for (int index = overflowLength - 1; index < text.Length; index++)
-                    {
-                        int nextDigit = text[index] - 48; // '0'
-                        if (nextDigit < 0 || nextDigit > 9)
-                        {
-                            value = parsedValue * sign;
-                            return true;
-                        }
-                        // If parsedValue > (int.MaxValue / 10), any more appended digits will cause overflow.
-                        // if parsedValue == (int.MaxValue / 10), any nextDigit greater than 7 or 8 (depending on sign) implies overflow.
-                        bool positive = sign > 0;
-                        bool nextDigitTooLarge = nextDigit > 8 || (positive && nextDigit > 7);
-                        if (parsedValue > int.MaxValue / 10 || parsedValue == int.MaxValue / 10 && nextDigitTooLarge)
-                        {
-                            value = default(int);
-                            return false;
-                        }
-                        parsedValue = parsedValue * 10 + nextDigit;
-                    }
-                }
-
-                value = parsedValue * sign;
-                return true;
+                return TryParseInt32(text, out value, out int bytesConsumed);
             }
 
             public static bool TryParseInt32(ReadOnlySpan<byte> text, out int value, out int bytesConsumed)
             {
-                if (text.Length < 1)
-                {
-                    bytesConsumed = 0;
-                    value = default(int);
-                    return false;
-                }
+                int textLength = text.Length;
+                if (textLength < 1) goto FalseExit;
 
-                int indexOfFirstDigit = 0;
-                int sign = 1;
-                if (text[0] == '-')
+                sbyte sign = 1;
+                int index = 0;
+                byte num = text[index];
```

Comment:
It will likely work better for this to be `int`, so that you pay for byte->int zero extension just once.